lib_java/src/main/java/catala/dates_calc/Date.java

```diff
@@ -0,0 +1,265 @@
+package catala.dates_calc;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Note that Date is purposefully non-final (we want to inherit from it in the
+ * java catala runtime, in order to add a marker interface)
+ */
+public class Date implements Comparable<Date> {
+
+    public enum Rounding {
+        ROUND_UP, ROUND_DOWN, ABORT_ON_ROUND
+    }
+
+    private final int year;
+    private final int month;
+    private final int day;
```

Comment:
```suggestion
    public final int year;
    public final int month;
    public final int day;
```
The runtime needs to access those. Some `get<Field>()` methods would also do the trick.